packages/cli/src/index.js

```diff
@@ -97,6 +106,78 @@ config({
   multiline: true,
 })
 
+const TELEMETRY_ENABLED =
+  telemetry !== 'false' && !process.env.REDWOOD_DISABLE_TELEMETRY
+
+async function main() {
+  // Start telemetry if it hasn't been disabled
+  if (TELEMETRY_ENABLED) {
+    try {
+      await startTelemetry()
+    } catch (error) {
+      console.error('Telemetry startup error')
+      console.error(error)
+    }
+  }
+
+  // Execute CLI within a span, this will be the root span
+  const tracer = trace.getTracer('redwoodjs')
+  await tracer.startActiveSpan('cli', async (span) => {
+    try {
+      // Run the command via yargs
+      await runYargs()
+
+      // Span housekeeping
+      span?.setStatus({ code: SpanStatusCode.OK })
+    } catch (error) {
+      const errorReferenceCode = uuidv4()
+      const errorMessage = [
+        error.stack ?? error.toString(),
+        '',
+        '-'.repeat(process.stdout.columns - 8),
+        '',
+        'Need help?',
+        ` - Not sure about something or need advice? Reach out on our ${terminalLink(
+          'Forum',
+          'https://community.redwoodjs.com/'
+        )}`,
+        ` - Think you've found a bug? Open an issue on our ${terminalLink(
+          'GitHub',
+          'https://github.com/redwoodjs/redwood'
+        )}`,
+        ` - Here's your unique error reference: '${errorReferenceCode}'`,
+      ].join('\n')
+      console.error(
+        boxen(errorMessage, {
+          padding: 1,
+          borderColor: 'red',
+          title: `Unexpected Error`,
+          titleAlignment: 'left',
+        })
+      )
+
+      recordTelemetryError(error)
+      recordTelemetryAttributes({ errorReferenceCode })
+      if (process.exitCode !== 0) {
+        process.exitCode ??= error.exitCode ?? 1
+      }
+
+      // Legacy telemetry
+      errorTelemetry(process.argv, error.message)
```

Comment:
Mostly just a note for others, this will be removed eventually, but till we've completely made the switch we'll keep it here.